### PR TITLE
Change Comments from `/*!` to `/**` to make possible jsdoc available

### PR DESCRIPTION
**Pull request by hasezoey**
**Summary**

This PR changes most `/*!` to `/**` to make jsdoc available for those that have descriptions / tags defined, also does:
- change some casing for `* ignore`
- change some `@ignore` to `ignore`
- change some comments from `/*!` to `//`
- add / change tags to be more in line with current implementations (for those that were `/*!` before)
- change all `@private` to `@api private`
- changes `api public` to `api private` for `const mongoose = module.exports = exports = new Mongoose({`, because this had no effect anyway (from what i could tell)
- `Model.prototype.$remove` is now public in the documentation
- `Model.db;` now properly appears as a static property in the documentation
- `ArraySubdocument.prototype.$parent` is now `api public` (because other aliases are)
- removes `@api` tags from jsdoc in `test/*` files

Things that were ignored:
- all that were just `ignore`
- all that said `module exports`
- all that said `inherits from` (assigning `.prototype` / `._proto_`)

### CODE REVIEW

File: lib/types/ArraySubdocument.js [hunk 1]
```diff
@@ -10,12 +10,14 @@ const utils = require('../utils');
...
 const documentArrayParent = require('../helpers/symbols').documentArrayParent;
 
-/*!
+/**
  * A constructor.
  *
  * @param {Object} obj js object returned from the db
  * @param {MongooseDocumentArray} parentArr the parent array of this document
  * @param {Boolean} skipId
+ * @param {Object} Fields
```
Comment (vkarpov15) [inline lib/types/ArraySubdocument.js:19]: Nit: lowercase F

**Review by vkarpov15 — APPROVED** [review]
Thanks!

File: lib/types/ArraySubdocument.js (hunk 1 above)
Comment (hasezoey) [inline lib/types/ArraySubdocument.js:19]:
```suggestion
 * @param {Object} fields
```

**Comment by hasezoey** [comment]
fixed with proper merge

**Pull request merged**